vault/eventbus/bus.go

```diff
@@ -203,8 +246,18 @@ func (node *asyncChanNode) Process(ctx context.Context, e *eventlogger.Event) (*
 		select {
 		case node.ch <- eventRecv:
 		case <-ctx.Done():
+			if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+				node.logger.Info("Node took too long to respond, closing")
+				node.Close()
+				return
+			}
 			return
 		case <-node.ctx.Done():
+			if errors.Is(node.ctx.Err(), context.DeadlineExceeded) {
+				node.logger.Info("Node took too long to respond, closing")
+				node.Close()
+				return
+			}
 			return
 		}
```

Comment:
nit: Could reduce the code duplication with a variable? And we can't really identify the subscriber, but it would be nice to include some identifying information for the event that got dropped.

```suggestion
		var timeout bool
		select {
		case node.ch <- eventRecv:
		case <-ctx.Done():
			timeout = errors.Is(ctx.Err(), context.DeadlineExceeded)
		case <-node.ctx.Done():
			timeout = errors.Is(node.ctx.Err(), context.DeadlineExceeded)
		}
		if timeout {
			node.logger.Info("Subscriber took too long to process event, closing", "ID", eventRecv.Event.ID())
			node.Close()
		}
```